routing/payment_lifecycle_test.go

```diff
@@ -698,7 +698,7 @@ func TestRouterPaymentStateMachine(t *testing.T) {
 			routes: []*route.Route{
 				shard, shard, shard, shard,
 			},
-			paymentErr: channeldb.ErrPaymentTerminal,
+			paymentErr: channeldb.FailureReasonPaymentDetails,
```

Comment:
Great 👍  I was wondering why the `paymentErr` was `ErrPaymentTerminal` in the commit this test was added, but I guess that was just to make the test pass before you fixed the logic?

Reply:
Yeah! Just to make sure we're actually hitting that error, then add the fix & update test.